README.md

```diff
@@ -59,58 +59,66 @@ solang compile --target solana flipper.sol
 
 Alternatively if you want to use the solang container, run:
 
+```bash
+docker run --rm -it -v $(pwd):/sources ghcr.io/hyperledger/solang compile -v -o /sources --target solana /sources/solana/flipper.sol
 ```
-docker run --rm -it -v $(pwd):/sources ghcr.io/hyperledger/solang compile -v -o /sources --target solana /sources/flipper.sol
+
+This generates a file called `flipper.json` and `flipper.so`. You can deploy the contract code using:
+
+```bash
+solana program deploy flipper.so
 ```
 
-A file called `flipper.abi` and `flipper.so`. Now install `@solana/solidity`:
+Now install `@project-serum/anchor`:
 
 ```
-npm install @solana/solidity
+npm install @project-serum/anchor
 ```
 
 Save the following to `flipper.js`:
 ```javascript
-const { Connection, LAMPORTS_PER_SOL, Keypair } = require('@solana/web3.js');
-const { Contract, Program } = require('@solana/solidity');
 const { readFileSync } = require('fs');
+const anchor = require('@project-serum/anchor');
 
-const FLIPPER_ABI = JSON.parse(readFileSync('./flipper.abi', 'utf8'));
+const IDL = JSON.parse(readFileSync('./flipper.json', 'utf8'));
 const PROGRAM_SO = readFileSync('./flipper.so');
 
 (async function () {
-    console.log('Connecting to your local Solana node ...');
-    const connection = new Connection('http://localhost:8899', 'confirmed');
+	const provider = anchor.AnchorProvider.env();
```

Comment:
Using `AnchorProvider.local()` or `new AnchorProvider()`

Reply:
If we use `AnchorProvider.local()` and if users want to use devnet/testnet rather than a local development node, then it is not clear how that switch can be made. Using this method, the url to the solana network is clear as day.

Reply:
I might be getting this wrong, but you can specify the url when you use `.local()`. The url points to either devenet or testnet.

Reply:
Like I said, if users want to use devnet or testnet, it is much clearer how they switch the url, if the url is in an environment variable rather than in code. The wallet has to be in the environment already. Having half the config in code and the other half in environment variables is not great.

Reply:
> Like I said, if users want to use devnet or testnet, it is much clearer how they switch the url, if the url is in an environment variable rather than in code. The wallet has to be in the environment already. Having half the config in code and the other half in environment variables is not great.

That makes sense. Keep in mind you can configure your wallet programmatically too.

Reply:
>  Keep in mind you can configure your wallet programmatically too.

How would we do that?

Reply:
```typescript
const keypair = Keypair.generate();
const wallet = new NodeWallet(keypair);
```
`NodeWalled` implements the wallet interface in Anchor.

Reply:
This is how we can read the key from a file:

```typescript
const keypair = Keypair.fromSecretKey(
      Buffer.from(
        JSON.parse(
          require("fs").readFileSync("payer.key", {
            encoding: "utf-8",
          })
        )
      )
    );
```

Reply:
That does not help us improve the example code.

Reply:
Please show me how the example can be done better. That's what I meant by "How would we do that?". Right now you're just giving me some code snippets (that I am already aware of).

I think you end up with more lines of code than what we have now, and I'm optimizing for code size. Also it's easy to switch wallet/url because it's very clearly marked in the environment variables, and not in the the middle of some javascript.

Either way, this will all go away with anchor tooling in place, so we shouldn't worry about it too much.